libr/util/idpool.c

```diff
@@ -163,6 +163,39 @@ R_API void* r_id_storage_get(RIDStorage* storage, ut32 id) {
 	return storage->data[id];
 }
 
+// usually these shall not be used
+// id-storage doesn't provide sorting things in specific order
+// there are only a very few reasonable usecases, such as cycling through RIODescs
+R_API bool r_id_storage_get_next(RIDStorage *storage, ut32 *id) {
+	r_return_val_if_fail (id && storage && storage->data, false);
+	// it does not indicate a bug, if the passed id goes beyond the boundaries of the storage
+	// therefor no r_return_val_if_fail for this case
+	if (storage->size <= *id) {
+		return false;
+	}
+
+	ut32 _id = *id;
+	do {
+		_id = (_id + 1) % storage->size;
+	} while (!storage->data[_id]);
```

Comment:
I haven't checked whether that's true or not, but this code assumes there will always be a NULL element in `storage->data`. Is that the case? Because if not, there could be an infinite loop.

Reply:
The code assumes, that there is at least 1 non-NULL element in the storage... but you're right, this can escalate to an endless loop (think about "r2 --") . Nice catch. I fixed that now